rclcpp/include/rclcpp/executors.hpp

```diff
@@ -26,6 +29,20 @@ namespace executors
 using rclcpp::executors::multi_threaded_executor::MultiThreadedExecutor;
 using rclcpp::executors::single_threaded_executor::SingleThreadedExecutor;
 
+template<typename FutureT>
+std::shared_future<FutureT> &
+spin_node_until_future_complete(
+  rclcpp::executor::Executor & executor, rclcpp::node::Node::SharedPtr & node_ptr,
+  std::shared_future<FutureT> & future)
+{
+  std::future_status status;
+  do {
+    executor.spin_node_some(node_ptr);
+    status = future.wait_for(std::chrono::seconds(0));
+  } while (status != std::future_status::ready && rclcpp::utilities::ok());
+  return future;
```

Comment:
Supporting chaining operators make usage much more compact. The other option I see is to have an additional or overridden `get` method which will do the background spinning. But I think this is cleaner and doesn't require modifying builtins.